frontend/packages/kubevirt-plugin/src/components/vm-snapshots/snapshot-row.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { TableData, TableRow, RowFunction } from '@console/internal/components/factory';
+import {
+  asAccessReview,
+  Kebab,
+  KebabOption,
+  ResourceLink,
+  Timestamp,
+} from '@console/internal/components/utils';
+import { deleteModal } from '@console/internal/components/modals';
+import { getName, getNamespace, dimensifyRow, getCreationTimestamp, Status } from '@console/shared';
+import { referenceFor } from '@console/internal/module/k8s';
+import { VirtualMachineSnapshotModel } from '../../models';
+import { isVMI } from '../../selectors/check-type';
+import { VMLikeEntityKind } from '../../types/vmLike';
+import { ValidationCell } from '../table/validation-cell';
+import {
+  VMSnapshotRowActionOpts,
+  SnapshotSimpleDataValidation,
+  VMSnapshotRowCustomData,
+} from './types';
+import { VMSnapshot } from '../../types';
+
+// TODO: add revertMenuAction once implemented
+
+const menuActionDelete = (
+  snapshot: VMSnapshot,
+  { withProgress }: { withProgress: (promise: Promise<any>) => void },
+): KebabOption => ({
+  label: 'Delete',
+  callback: () =>
+    withProgress(
+      deleteModal({
+        kind: VirtualMachineSnapshotModel,
+        resource: snapshot,
+      }),
+    ),
+  accessReview: asAccessReview(VirtualMachineSnapshotModel, snapshot, 'delete'),
+});
+
+const getActions = (
+  snapshot: VMSnapshot,
+  vmLikeEntity: VMLikeEntityKind,
+  opts: VMSnapshotRowActionOpts,
+) => {
+  if (isVMI(vmLikeEntity)) {
+    return [];
+  }
+  const actions = [menuActionDelete];
+  return actions.map((a) => a(snapshot, opts));
+};
+
+export type VMSnapshotSimpleRowProps = {
+  data: VMSnapshot;
+  validation?: SnapshotSimpleDataValidation;
+  columnClasses: string[];
+  actionsComponent: React.ReactNode;
+  index: number;
+  style: object;
+};
+
+export const SnapshotSimpleRow: React.FC<VMSnapshotSimpleRowProps> = ({
+  data: snapshot,
+  validation = {},
+  columnClasses,
+  actionsComponent,
+  index,
+  style,
+}) => {
+  const dimensify = dimensifyRow(columnClasses);
+  const name = getName(snapshot);
+  const namespace = getNamespace(snapshot);
+
+  return (
+    <TableRow id={snapshot?.metadata?.uid} index={index} trKey={name} style={style}>
+      <TableData className={dimensify()}>
+        <ValidationCell validation={validation.name}>
+          <ResourceLink
+            kind={referenceFor(VirtualMachineSnapshotModel)}
+            namespace={namespace}
+            name={name}
+          />
+        </ValidationCell>
+      </TableData>
+      <TableData className={dimensify()}>
+        <ValidationCell validation={validation.date}>
```

Comment:
creationTimestamp ? maybe it would be easier to remove the validations after all